tests/unit/htdocs/class/logger/XoopsFileLoggerTest.php

```
@@ -0,0 +1,353 @@
+
+    #[Test]
+    public function aNewlineInTheUriCannotForgeAnEntry(): void
+    {
+        $_SERVER['REQUEST_URI'] = "/index.php?x=1\n[2026-01-01 00:00:00] messages.ERROR FORGED";
+        $logger                 = $this->makeLogger();
+        $logger->log('warning', 'a message', ['channel' => 'messages']);
+
+        // Whatever currentUri() yields, sanitize() strips control characters, so the entry
+        // is a header plus one body line and nothing more.
+        self::assertSame(2, substr_count($this->readLog(), "\n"));
+    }
```

PHPMD (2.15.0)

[error] aNewlineInTheUriCannotForgeAnEntry accesses the super-global variable $_SERVER. (undefined)

(Superglobals)